apps/rush-lib/src/cli/actions/PublishAction.ts

```diff
@@ -453,4 +472,28 @@ export class PublishAction extends BaseRushAction {
 
     return `${name}-${project.packageJson.version}.tgz`;
   }
-}
\ No newline at end of file
+
+  private _manageLooseVersions(action: 'lock' | 'unlock'): void {
+    const projectsByName: Map<string, RushConfigurationProject> = this.rushConfiguration.projectsByName;
+
+    for (const [, project] of projectsByName.entries()) {
+      const editor: PackageJsonEditor = project.packageJsonEditor;
+
+      const dependencies: ReadonlyArray<PackageJsonDependency> = editor.dependencyList;
+
+      for (const dependency of dependencies) {
+        const localProject: RushConfigurationProject | undefined = projectsByName.get(dependency.name);
+
+        if (localProject) {
+          if (action === 'lock') {
+            dependency.setVersion(localProject.packageJson.version);
+          } else {
+            dependency.setVersion('*');
+          }
+        }
+      }
+
+      editor.saveIfModified();
```

Comment:
The `PackageJsonEditor` causes a lot of shuffle in project files. Is there any way to make it less intrusive on existing `package.json` file organization?

Reply:
You mean that it moves things around in `package.json` files? I think `jju` supports that, but we might need to plumb something through. @octogonz would know better than me.

Reply:
Yes, it moves things around quite a bit. Now, having it 'standardize' the order is good long-term, but disruptive initially.